cmd/runner/config.go

```diff
@@ -8,38 +8,26 @@ import (
 	"github.com/CircleCI-Public/circleci-cli/api/runner"
 )
 
-type AgentConfig struct {
-	API    APIConfig    `yaml:"api"`
-	Runner RunnerConfig `yaml:"runner"`
-}
-
-func NewAgentConfig(t runner.Token) *AgentConfig {
-	return &AgentConfig{
-		API: APIConfig{
+func generateConfig(t runner.Token, w io.Writer) (err error) {
+	return yaml.NewEncoder(w).Encode(&agentConfig{
+		API: apiConfig{
 			AuthToken: t.Token,
 		},
-		Runner: RunnerConfig{
-			Name:                    t.Nickname,
-			ResourceClass:           t.ResourceClass,
-			CommandPrefix:           []string{"/opt/circleci/launch-task"},
-			WorkingDirectory:        "/opt/circleci/workdir/%s",
-			CleanupWorkingDirectory: true,
+		Runner: runnerConfig{
+			Name: t.Nickname,
```

Comment:
I think the token nickname should be treated as distinct from the runner name - as we expect tokens to be shared across machines but require runner names to be unique